connector/countconnector/counter.go

```diff
@@ -117,3 +125,24 @@ func (c *counter[K]) appendMetricsTo(metricSlice pmetric.MetricSlice) {
 		}
 	}
 }
+
+func getNestedAttribute(attr string, depth int, attributes pcommon.Map) (pcommon.Value, bool) {
+	logger := zap.NewExample()
+	defer logger.Sync()
+	left, right, _ := strings.Cut(attr, ".")
+	av, ok := attributes.Get(left)
+	if ok && depth > 0 {
+		if av.Type().String() == "Slice" {
+			logger.Warn("[connector/count] cannot use Type Slice as an attribute source", zap.String("attribute", attr))
```

Comment:
I don't think we should instantiate a logger locally like this. It's not clear to me that we really need to log anything in this case anyways. Seems we can just consider it a thing we can't drill into any further, same as if we found any other data type besides a map.